### ISSUE TITLE
Feature Request: Training Lora on inpainting models.

### ISSUE THREAD

**Issue by miniwa** [kohya-ss/sd-scripts#502]
As a developer, one of the challenges I've encountered is that lora networks, when trained on standard models, tends to underperform when used with inpainting models.

The possible solution: Allowing creators to publish two variants of a Lora for the same concept - a "normal" Lora and an "inpainting" Lora. This would empower users with the flexibility to choose the best-suited model for their specific applications, thus improving overall user experience and satisfaction.

cloneofsimo has implemented this in his repo. Probably can use the same method as he does. Some examples below:
https://github.com/cloneofsimo/lora/blob/bdd51b04c49fa90a88919a19850ec3b4cf3c5ecd/lora_diffusion/dataset.py#L268
https://github.com/cloneofsimo/lora/blob/master/training_scripts/inpainting_example.sh

Thank you for all your hard work!

**Comment by ceswap** [comment]
Yes, that would be really nice to have +1

**Comment by Vadmidin** [comment]
Yes, please implement!

**Comment by miniwa** [comment]
> Regarding the first issue, similar functionality is also required for ControlNet and mask loss. Thus, I plan to implement this in the near future. Once that is done, I hope to further consider this issue.

These are great features on their own. :)

**Comment by julien-blanchon** [comment]
Any news ?

**Comment by PMykhailo** [comment]
How do you even train lora for img2img? I searched for methods, but find none. I'm interesting particularly in inpainting

**Comment by PMykhailo** [comment]
Maybe this will help some of you all. Nut tested it myself, yet https://github.com/huggingface/diffusers/tree/main/examples/research_projects/dreambooth_inpaint
